sdk/python/feast/usage.py

```diff
@@ -73,14 +90,28 @@ def usage_id(self) -> Optional[str]:
             return os.getenv("FEAST_FORCE_USAGE_UUID")
         return self._usage_id
 
-    def log(self, function_name: str):
+    async def _send_request(self, json: Dict):
+        requests.post(USAGE_ENDPOINT, json=json)
+
+    async def _send_usage_request(self, json):
+        try:
+            task = asyncio.ensure_future(self._send_request(json))
+            if self._is_test:
+                await task
+        except Exception as e:
+            if self._is_test:
+                raise e
```

Comment:
Why are we doing this async?

Reply:
Seemed like this was adding significant latency for the requests when they fire.

Verified that this was saving ~100-200ms for when the events are logged.